app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/event_parser.rb

```diff
@@ -0,0 +1,42 @@
+module ManageIQ::Providers::IbmCloud::PowerVirtualServers::CloudManager::EventParser
+  def self.event_to_hash(event, ems_id)
+    event_hash = {
+      :event_type => "#{event[:resource]}.#{event[:action]}",
+      :source     => "IBMCloud-PowerVS",
+      :ems_id     => ems_id,
+      :ems_ref    => event[:eventID],
+      :timestamp  => event[:time],
+      :full_data  => event,
+      :username   => event[:user][:email]
```

Comment:
Not sure if `event[:user]` can ever be nil, but if it can this will fail

`event.dig(:user, :email)` is a good way to handle this condition